### DISCUSSION TITLE
Building a great select widget for Piccolo Admin

### DISCUSSION THREAD

**Discussion by dantownsend** [piccolo-orm/piccolo#372] [General]
We revamped the foreign key selector in Piccolo Admin, and released it this week.

![hybrid_select_widget](https://user-images.githubusercontent.com/350976/145672911-ab51e228-b6e9-4014-b8ab-481a9277bbfb.jpeg)

It was surprisingly tricky making this piece of UI. Here's a video about it, and the thought process behind it:

https://www.youtube.com/watch?v=h819CBKIqKI

And an article about it:

https://piccolo-orm.com/blog/building-a-great-select-widget/

**Comment by sinisaos** [comment]
@dantownsend I've already commented on this widget, but again this widget looks and works great.

**Reply by dantownsend** [reply]
Thanks!
